tests/spec/fs.rename.spec.js

```diff
@@ -151,4 +151,35 @@ describe('fs.rename', function() {
       });
     });
   });
+
+  it('(promise version) should rename an existing file', function(done) {
+    var complete1 = false;
+    var complete2 = false;
+    var fs = util.fs();
+
+    function maybeDone() {
```

Comment:
Instead of doing things this way, with a `done` callback and a bunch of booleans to track what is and isn't finished, we can just rely on Promises to do it.  Change your test code like this:

1. Remove the `done` callback arg on line 155.  We won't use it (see 2.)
1. Add a `return` to the front of line 167 below, so you are returning your promise.  This will tell it when the test is done.
1. See my inline changes below